### PR TITLE
ROS form user interface update

### PR DESCRIPTION
**Pull request by juggernautsei**
If we are going with a new interface some of the old forms need a new look also.

### CODE REVIEW

**Comment by robertdown** [comment]
Agreed that we should update the look of forms.

We should hold off until we have a standard template to follow. Really just need some OpenEMR-specific documentation on Bootstrap’s `container` `row` and `col-*` classes as well as the form classes.

File: interface/forms/ros/C_FormROS.class.php [hunk 1]
```diff
@@ -1,30 +1,47 @@
...
 class C_FormROS extends Controller {
 
-	var $template_dir;
+	public $template_dir;
 	
-    function __construct($template_mod = "general") {
+    public function __construct($template_mod = "general") {
     	parent::__construct();
-    	$returnurl = 'encounter_top.php';
+    	$returnurl = $GLOBALS['concurrent_layout'] ? 'encounter_top.php' : 'patient_encounter.php';
```
Comment (bradymiller) [inline interface/forms/ros/C_FormROS.class.php]: revert this change. the concurrent_layout global has been removed.

File: interface/forms/ros/C_FormROS.class.php [hunk 2]
```diff
@@ -1,30 +1,47 @@
...
+ */
+ 
 require_once ($GLOBALS['fileroot'] . "/library/classes/Controller.class.php");
 require_once ($GLOBALS['fileroot'] . "/library/forms.inc");
 require_once("FormROS.class.php");
 
 class C_FormROS extends Controller {
 
-	var $template_dir;
+	public $template_dir;
```
Comment (bradymiller) [inline interface/forms/ros/C_FormROS.class.php]: Why did you need to change all the vars and functions to public?

Comment (juggernautsei) [inline interface/forms/ros/C_FormROS.class.php]: I changed them because of the course on OOP that I am in. Everything we write for our classes is labeled public private or protected. And I was told that var was the old way. So, I was going to embark on updating how the classes are referenced. Since I saw it in the new code for the fee sheet. I thought it would be a two for one while updating the look of the ROS that I was asked to do. I thought I would sneak the changes in those classes in at this time. 
Even though at this time it makes no difference in the operation of the class. One of the other things that I learned was auto instantiation. That could cut the size of the FormROS in half or better if I can get it working. I'm just trying find something useful to do to be a part of the bigger picture.

File: interface/forms/ros/C_FormROS.class.php (hunk 1 above)
Comment (juggernautsei) [inline interface/forms/ros/C_FormROS.class.php]: That change is  not in my code. Hmmmm

File: interface/forms/ros/C_FormROS.class.php (hunk 2 above)
Comment (robertdown) [inline interface/forms/ros/C_FormROS.class.php]: I've never used `var` myself, always been `public`, `private`, or `protected`. I like this change

File: interface/forms/ros/C_FormROS.class.php (hunk 1 above)
Comment (bradymiller) [inline interface/forms/ros/C_FormROS.class.php]: Just means your code is a bit out of date. The concurrent_layout was stripped out of the codebase a couple weeks ago. Will need to change it to:
``````
$returnurl = 'encounter_top.php';

File: interface/forms/ros/FormROS.class.php [hunk 3]
```diff
@@ -1,5 +1,22 @@
...
+ * of the License, or (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://opensource.org/licenses/gpl-license.php>;.
+ *
+ * @package OpenEMR
+ * Sherwin Gaddis <sherwingaddis@gmail.com>
```
Comment (bradymiller) [inline interface/forms/ros/FormROS.class.php]: Need to place @author tag here(also do for your other headers below):
``````
* @author Sherwin Gaddis <sherwingaddis@gmail.com>

**Comment by bradymiller** [comment]
@robertdown 
Would this be a good starting point/entry/example into introducing bootstrap into the main codebase?

**Comment by juggernautsei** [comment]
If you give me what you want me to do, I will build it into the form. I have used the bootstrap before. Just let me know.

**Comment by bradymiller** [comment]
@robertdown , Should Sherwin include the bootstrap css and js on these files and go from there?

**Comment by mdsupport** [comment]
Our 'standard' theme starts like this:

```
@import "style_light.css";
@import "../../public/assets/bootstrap-3-3-4/dist/css/bootstrap.min.css";
@import "../../public/assets/font-awesome-4-6-3/css/font-awesome.min.css";

body {
  font-size: 10px; !important;
}
.css_button_small, ul.tabNav a {

```

**Comment by juggernautsei** [comment]
I will make these adjustments to the form for the styling

File: interface/forms/ros/C_FormROS.class.php [hunk 4]
```diff
@@ -1,30 +1,47 @@
...
+ * of the License, or (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://opensource.org/licenses/gpl-license.php>;.
+ *
+ * @package OpenEMR
+ * Sherwin Gaddis <sherwingaddis@gmail.com>
```
Comment (juggernautsei) [inline interface/forms/ros/C_FormROS.class.php]: I finally got a chance to get back to this form and I have looked at some of the new code. Here is my two cents. 

In the smarty template there is a line that is 
 {php}html_header_show();{/php}
which as you know sets the theme that is going to be used. My question is there a way to deliver the bootstrap location the same way? That way it is universally distributed like the theme. Then we all should be able to build or rebuild on the same framework. 
Am I talking out of my head or saying what is already in the works?

File: interface/forms/ros/FormROS.class.php [hunk 5]
```diff
@@ -25,158 +42,158 @@ class FormROS extends ORDataObject {
...
-	var $diabetes = "N/A";
-	var $abnormal_blood = "N/A";
-	var $anemia = "N/A";
-	var $fh_blood_problems = "N/A";
-	var $bleeding_problems = "N/A";
-	var $allergies = "N/A";
-	var $frequent_illness = "N/A";
-	var $hiv = "N/A";
-	var $hai_status = "N/A";
+	public $id;
```
Comment (juggernautsei) [inline interface/forms/ros/FormROS.class.php]: @mdsupport ,
I was revisiting this change and I like the suggest of function call to clean up the original file. But you also mention converting to a LBF. Wouldn't make it an LBF would make controlling the "layout". LBF's don't always turn out as desired in my experience unless the form is done like the history or portions of the demographics that are stored in the LBF table.
The the LBF be controlled with CSS like using the js?

**Comment by robertdown** [comment]
Closing this since outdated PR
